### ISSUE TITLE
Support constructing new STRUCTs (i.e., writing nested data)

### ISSUE THREAD

**Issue by miguno** [confluentinc/ksql#2147]
TL;DR: The goal is to support creating new STRUCTs in your queries.

# Background

KSQL already supports *reading* nested data (JSON, Avro) via the STRUCT type:

- Accessing (reading) existing nested data in your queries.  For example, you can access a nested field via the `myStruct->myField` syntax.
- "Limited write" support: you can copy nested data -- i.e., a STRUCT -- as-is (unmodified) from an input stream/table to an output/stream table.

However, KSQL does not yet support:

- Creating a new STRUCT from the input stream/table (i.e., in a CSAS/CTAS statement).

# Example

```sql
-- We start with a non-nested stream.
-- The goal is to turn the flat street and zip fields into a nested address STRUCT.
CREATE STREAM orders (item VARCHAR, quantity INTEGER, street VARCHAR, zip INTEGER) WITH (...);

-- The following is not yet supported!
-- Note: The syntax below is for illustrative purposes and does not preclude
-- the design discussion we need to have.
CREATE STREAM nested_orders AS
  SELECT
    item, quantity,
    STRUCT(street, zip) AS address1,
    STRUCT<renamedStreet VARCHAR, renamedZip INTEGER>(street, zip) AS address2,
    STRUCT<VARCHAR, LONG>(street, zip) AS address3
  FROM orders;
```

**Comment by miguno** [comment]
+1 from a user in an offline conversation

**Comment by rmoff** [comment]
+2 from two separate user interactions this week at meetups

**Comment by Recodify** [comment]
+1 https://stackoverflow.com/questions/1602934/check-if-a-given-key-already-exists-in-a-dictionary

Thanks for the response @rmoff

**Comment by urspralini** [comment]
Looking forward for this support.

**Comment by brunodomenici** [comment]
Upvoting...we need that too :)

**Comment by deadzg** [comment]
+1
Looking forward for the support

**Comment by rsaltrelli** [comment]
+1

**Comment by yunhappy** [comment]
+1
Looking forward for the support

**Comment by muirandy** [comment]
+1

**Comment by ishrat-pge** [comment]
+1

**Comment by chrisngene** [comment]
+1

**Comment by agavra** [comment]
Thanks everyone for upvoting this! I just merged this feature request and it should be available in the next release 😄
